backend/de.metas.edi/src/main/java/de/metas/edi/api/impl/EDIDocumentBL.java

```diff
@@ -102,21 +103,34 @@ public boolean updateEdiEnabled(@NonNull final I_EDI_Document_Extension document
 		if (docStatus.isReversed() && document.isEdiEnabled())
 		{
 			loggable.addLog("DocStatus={} is reversed; => update IsEdiEnabled to false", docStatus);
-			document.setIsEdiEnabled(false);
+			setEdiEnabled(document, false);
 			return document.isEdiEnabled();
 		}
 
 		if (document.getReversal_ID() > 0 && document.isEdiEnabled())
 		{
 			loggable.addLog("Reversal_ID={} (i.e. >0); => update IsEdiEnabled to false", docStatus);
-			document.setIsEdiEnabled(false);
+			setEdiEnabled(document, false);
 			return document.isEdiEnabled();
 		}
 
 		logger.debug("return non-updated isEdiEnabled={}", document.isEdiEnabled());
 		return document.isEdiEnabled();
 	}
 
+	@Override
+	public void setEdiEnabled(@NonNull final I_EDI_Document_Extension document, final boolean isEdiEnabled)
+	{
+
+		if(!isEdiEnabled && EDIExportStatus.isInProgressOrSend(EDIExportStatus.ofCode(document.getEDI_ExportStatus())))
```

Comment:
we would need a similar (or the same?) guard also for the case when we enable the EDI.
what if the edi status is in progress or sent and we are asked to enable the EDI (maybe it was alrady enabled?) and now we are moving back the status to Pending?